pkg/client/client.go

```diff
@@ -341,25 +363,29 @@ func (c *dyskclient) pageblob_get(d *Dysk) (*storage.Blob, error) {
 	containerPath = containerPath[1:]
 	blobContainer := blobClient.GetContainerReference(containerPath)
 
-	exists, err := blobContainer.Exists()
-	if nil != err {
-		return nil, err
-	}
-	if !exists {
-		return nil, fmt.Errorf("Container at %s does not exist", d.Path)
+	// we can not really before the "exist" checks
+	// using sas -- since sas will limit access to the file only
+	if !c.usingSas {
+		exists, err := blobContainer.Exists()
+		if nil != err {
+			return nil, err
+		}
+		if !exists {
+			return nil, fmt.Errorf("Container at %s does not exist", d.Path)
+		}
 	}
-
 	pageBlobName := path.Base(d.Path)
 	pageBlob := blobContainer.GetBlobReference(pageBlobName)
 
-	exists, err = pageBlob.Exists()
-	if nil != err {
-		return nil, err
-	}
-	if !exists {
-		return nil, fmt.Errorf("Blob at %s does not exist", d.Path)
+	if !c.usingSas {
```

Comment:
Can we check if the page blob exists using sas?

Reply:
We should be able to. I  was talking with @ehotinger to add `dyskctl mount -a .. -sas ...`. The idea is not break existing interface nor cli since we have users already using it